### ISSUE TITLE
Docker server pulls irrelevant images when the version of the Docker image is not set into the task definition

### ISSUE THREAD

**Issue by radenui** [aws/amazon-ecs-agent#100]
Hi,

On amazon linux using Docker 1.6.2, it seems like Docker pulls random images when no version of the Docker image is set into the task definition.
For instance:

- If the task definition references the image "registry", Docker will pull unusable images until the devicemapper directory fills the whole filesystem
- If I try to pull the image "registry" directly, if works
- If I reference "registry:latest" instead of "registry", it works.

Any idea what's happening ?

Thanks

**Comment by euank** [comment]
Hi, radenue, it sounds to me like you're encountering the bug fixed in #75.
That fix hasn't been included in a release yet. Then next release should include it.

The bug that PR fixes is that an image without a tag specified had the same behavior as `docker pull -a` &mdash;  `docker pull -a registry` in your case &mdash; which results in it downloading all of the tags of the given image.

As you note, adding the `:latest` tag explicitly to your task definition works around this issue.

Thank you for reporting this and sorry for the inconvenience this bug has no doubt caused.

**Comment by radenui** [comment]
Ok, thanks, I hadn't seen this one

**Comment by tj** [comment]
Related is that if you accidentally get the tag wrong ECS seems to hang indefinitely instead of producing an error

**Comment by euank** [comment]
Sorry for the slow response, @tj. Can you give me a few more details about what you mean?

I tried to reproduce this by registering and running the following task definition:
```
{
	"family": "sleep-badtag",
	"containerDefinitions": [{
		"image": "busybox:badtag",
		"name": "bb",
		"memory": 10,
		"command": ["sleep", "5"]
	}]
}
```

Running it on both the v1.1.0 and current dev agent versions resulted in the task being moved to stopped with a reason (visible when describing it) of "CannotPullContainerError: Tag badtag not found in repository busybox".

I have, in the past, ended up with Docker hanging on some pulls (see issue https://github.com/docker/docker/issues/12823) which can hold up the agent significantly, but manually running `docker pull` for the image+tag it's on also hangs. Perhaps that's what you encountered?

Any additional info would be appreciated.
Thanks,
Euan

**Comment by tj** [comment]
Just did it again by accident! I guess ecs-agent just keeps trying, but I do get a bunch of Reason:CannotPulledContainerError errors in the logs, just non-obvious from the UI, it keeps trying over and over

**Comment by samuelkarp** [comment]
Released with [v1.2.0](https://github.com/aws/amazon-ecs-agent/releases/tag/v1.2.0).
